libexec/rtld-cheri-elf/rtld.c

```diff
@@ -4710,48 +4710,109 @@ tls_get_addr_common(intptr_t **dtvp, int index, size_t offset)
 #if defined(__aarch64__) || defined(__arm__) || defined(__mips__) || \
     defined(__powerpc__) || defined(__riscv__)
 
+/*
+ * Return pointer to allocated TLS block
+ */
+static void *
+get_tls_block_ptr(void *tcb, size_t tcbsize)
+{
+    size_t extra_size, post_size, pre_size, tls_block_size;
+    size_t tls_init_align;
+
+    tls_init_align = MAX(obj_main->tlsalign, 1);
+
+    /* Compute fragments sizes. */
+    extra_size = tcbsize - TLS_TCB_SIZE;
+#if defined(__aarch64__) || defined(__arm__)
+    post_size =  roundup2(TLS_TCB_SIZE, tls_init_align) - TLS_TCB_SIZE;
+#else
+    post_size = 0;
+#endif
+    tls_block_size = tcbsize + post_size;
+    pre_size = roundup2(tls_block_size, tls_init_align) - tls_block_size;
+
+    return ((char *)tcb - pre_size - extra_size);
+}
+
 /*
  * Allocate Static TLS using the Variant I method.
+ *
+ * To handle all above requirements, we setup the following layout for 
```

Comment:
The requirements aren't listed here (they are in the libc source), so I would perhaps just add a pointer to the comments in lib/libc/gen/tls.c.